### PR TITLE
Rework combat floodlights into something simpler

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Make floodlight combat simpler. Just wrench it, click it, and tada, you got lights. It is powered by a cell (a standard cell will hold 15 minutes), that you can change with a crowbar

## Why It's Good For The Game

Combat floodlight were completly unused mainly because the system is highly complex, for absolutly no reason. This make it simpler to use

## Changelog
:cl:
refactor: Rework combat floodlights into something simpler
/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
So they are no longer powered by areas? Or can they still be powered by areas?

**Comment by BraveMole** [comment]
> So they are no longer powered by areas? Or can they still be powered by areas?

No, only cells

File: code/game/objects/machinery/floodlight.dm [hunk 1]
```diff
@@ -80,226 +80,131 @@
 	set_light(brightness_on)
 
 /obj/machinery/floodlightcombat
-	name = "Armoured floodlight"
+	name = "armoured floodlight"
```
Comment (hyper2snyper) [inline code/game/objects/machinery/floodlight.dm:83]: to be 'armored' or to be armoured'
color or colour
grey or gray

**Review by hyper2snyper — COMMENTED** [review]
Looks fine to me.

**Comment by hyper2snyper** [comment]
Can you explain that you are making the req purchasable one twice as expensive.

File: code/game/objects/machinery/floodlight.dm (hunk 1 above)
Comment (BraveMole) [reply code/game/objects/machinery/floodlight.dm:83]: What is the issue? Both are acceptable

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/machinery/floodlight.dm (hunk 1 above)
Comment (hyper2snyper) [reply code/game/objects/machinery/floodlight.dm:83]: none, just musing cuz its funny.

File: code/game/objects/machinery/floodlight.dm [hunk 2]
```diff
@@ -80,226 +80,131 @@
...
+	GLOB.nightfall_toggleable_lights += src
+
+/obj/machinery/floodlightcombat/Destroy()
+	QDEL_NULL(cell)
+	GLOB.nightfall_toggleable_lights -= src
+	return ..()
+
+
+/obj/machinery/floodlightcombat/examine(mob/user)
+	..()
```
Comment (hyper2snyper) [inline code/game/objects/machinery/floodlight.dm:109]: any reason you didnt do `. = ..()` ?

File: code/game/objects/machinery/floodlight.dm [hunk 3]
```diff
@@ -80,226 +80,131 @@
...
+	user.put_in_hands(cell)
+	cell = null
+	turn_light(user, FALSE, forced = TRUE)
 
 /obj/machinery/floodlightcombat/process()
-	if(powered(LIGHT))
-		use_power(light_on ? active_power_usage : idle_power_usage, LIGHT)
-		machine_stat &= ~NOPOWER
+	cell.charge -= energy_consummed
+	if(cell.charge <= 0)
```
Comment (hyper2snyper) [inline code/game/objects/machinery/floodlight.dm:148]: early return?

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

**Comment by Derrick2Electricboogaloo** [comment]
> 

Sounds neat will give it a try.

**Pull request merged**